app/models/entity.rb

```diff
@@ -0,0 +1,122 @@
+class Entity
+
+  #TODO LOG
+  #TODO Error handling
+
+  def get_connec_entities(client, last_synchronization)
+    Rails.logger.debug "Fetching Connec! #{self.class.name.pluralize}"
+
+    entities = []
+
+    # Fetch first page
+    if last_synchronization.blank?
+      response = client.get("/#{self.connec_entity_name.downcase.pluralize}")
+    else
+      query_param = URI.encode("$filter=updated_at >= #{last_synchronization.updated_at}")
+      response = client.get("/#{self.connec_entity_name.downcase.pluralize}?#{query_param}")
+    end
+
+    response_hash = JSON.parse(response.body)
+    entities << response_hash["#{self.connec_entity_name.downcase.pluralize}"]
+
+    # Fetch subsequent pages
+    while response_hash['pagination'] && response_hash['pagination']['next']
+      # ugly way to convert https://api-connec/api/v2/group_id/organizations?next_page_params to /organizations?next_page_params
+      next_page = response_hash['pagination']['next'].gsub(/^(.*)\/#{self.name.downcase.pluralize}/, '/#{self.name.downcase.pluralize}')
+      response = client.get(next_page)
+      response_hash = JSON.parse(response.body)
+      entities << response_hash["#{self.connec_entity_name.downcase.pluralize}"]
+    end
+
+    entities.flatten
+  end
+
+  #TODO Pagination
+  def get_external_entities(client, last_synchronization)
+    Rails.logger.debug "Fetching external #{self.external_entity_name}"
+    # if last_synchronization
+      # Cannot get the get_updated query to work
+      # client.get_updated('Account', last_synchronization.updated_at, Time.now)
+      # client.query('select Id, Name from Account ORDER BY Name')
+    # else
+      fields = self.mapping.values.join(', ')
+      entities = client.query("select Id, LastModifiedDate, #{fields} from #{self.external_entity_name} ORDER BY LastModifiedDate DESC")
+      entities = entities.to_a
+
+      index = 0
```

Comment:
The idea is that the array is sorted so we just have to find the first entity with a LastModifiedDate >= last_synchronization.updated_at.
There's probably a prettier way to do it, but i'm affraid select will uselessly iterate over the whole array (which can be quite long)